### ISSUE TITLE
Docker build for new 1809 images / OS build 10.0.17763.1

### ISSUE THREAD

**Issue by StefanScherer** [appveyor/ci#2676]
Any plans when you will support a Docker build with Windows Server 2019 which has been released this week?

The image names have changed for the Windows base images:

old:
- microsoft/windowsservercore:ltsc2016
- microsoft/nanoserver:sac2016

new:
- mcr.microsoft.com/windows/servercore:ltsc2019
- mcr.microsoft.com/windows/nanoserver:1809

And it is still the same issue that we need to have an agent with the same kernel version 10.0.17763.1 to run Docker builds for these new images.

**Comment by FeodorFitsner** [comment]
Hi Stefan,

We've already started working on Windows Server 2019 image! :)

Is there WSL and LCOW available on WS 2019? If yes, could you share some info/links to read more?

Thank you!

**Comment by StefanScherer** [comment]
WSL is available in WS 2019.
```
Install-WindowsFeature Microsoft-Windows-Subsystem-Linux
```
See also https://docs.microsoft.com/en-us/windows/wsl/install-on-server
But I haven't installed a Linux distro yet, too many other things to try ;-)

LCOW seems to work when you use the latest Docker EE 18.03 available and install the LinuxKit kernel
This code snippet might help https://github.com/StefanScherer/windows-docker-machine/blob/59b8dd940e560a5fbe2a101c5cde3fbaae65269f/scripts/create-machine.ps1#L35-L48
I have tried it with some Insider builds a few weeks ago.

```
Install-Package -Name docker -ProviderName DockerMsftProvider -RequiredVersion 18-03-1-ee-3
```

**Comment by StefanScherer** [comment]
Yes, the default has changed in moby/moby aka Docker engine. https://github.com/linuxkit/lcow/commit/5441411ad41bbc51eb2de7a1b498d885975864cc
But there is no newer lcow release yet so I had to move the file manually.

**Comment by FeodorFitsner** [comment]
Actually, looks like Azure still has "Preview" image for Windows Server 2019 :( as I'm getting this while trying to run new nano container:

```
PS C:\Users\appveyor> docker run mcr.microsoft.com/windows/nanoserver:1809
Unable to find image 'mcr.microsoft.com/windows/nanoserver:1809' locally
1809: Pulling from windows/nanoserver
C:\Program Files\Docker\docker.exe: a Windows version 10.0.17763-based image is incompatible with a 10.0.17666 host.
See 'C:\Program Files\Docker\docker.exe run --help'.
```

**Comment by StefanScherer** [comment]
The new VM images should show up in Azure next week. I have built a custom image in Azure for Chocolatey Fest. 

The `kernel` was needed for the nightly Docker engine. Good to know that 18.03 EE does not have this latest change yet.

**Comment by StefanScherer** [comment]
The RequiredVersion can be shortened to 18.03 to get the latest if you like.

**Comment by StefanScherer** [comment]
In the meantime there is also Docker EE 18.09.0 which is already available in the Azure 2019 with Containers templates.

**Comment by sixeyed** [comment]
The Azure image is called *Windows Server 2019 Datacenter with Containers* - it has Docker installed and the latest Server Core and Nano base images already pulled.

**Comment by darrenfurr** [comment]
Server 2019 with Containers (Docker 18.09) & using LCOW on an Azure VM to an Azure ACR did not work for me. See this issue: https://github.com/docker/for-win/issues/2623

**Comment by FeodorFitsner** [comment]
@darrenfurr Thanks for sharing your experience! That's sad to know though. When I was playing with LCOW on Windows Server 2019 last time I didn't test push.

**Comment by IanBFarrell** [comment]
Has someone confirmed that the push worked with 18.03? because I have EOF errors with 18.09

**Comment by FeodorFitsner** [comment]
@IanBFarrell What AppVeyor image do you use?

**Comment by IanBFarrell** [comment]
Sorry, I am not using a AppVeyor image. This issue was linked from another issue on the Docker for-win issues and I followed it. If this issue is specific to AppVeyor images then I am sorry for bothering you.

Issue that brought me here: [docker/for-win#2623](https://github.com/docker/for-win/issues/2623)

**Comment by specialwork** [comment]
Hi together, 
does somebody know how to expose docker daemon without TLS on a Windows Server 2019. I know how to do that on Windows 10 but is there a durable way to do this on Windows Server 2019?

Environment: 
Windows Server 2019 VM on Azure (Docker is running), Ubuntu WSL with (Docker is running, but cannot connect to the docker daemon)

**Comment by StefanScherer** [comment]
Is there an ETA to have Windows Server 2019 build agents?

**Comment by IlyaFinkelshteyn** [comment]
We already have image `Windows Server 2019` running on Azure and provide access to it per request. However we installed Docker Engine Enterprise like described [here](https://docs.docker.com/install/windows/docker-ee/) + LCOW. People who tried it were not satisfied because it lucks a lot of features so we re-working it now with Docker Desktop or Docker for Windows (sorry if I am wrong with terminology) which works with MobyLinux VM. I plan to release it by the end of month.

**Comment by IlyaFinkelshteyn** [comment]
`Windows Server 2019` image preview is available. It has both Docker CE to run Linux Docker instances, as well as WSL. Usage example is [here](https://ci.appveyor.com/project/appveyor-tests/win2k19-azure/builds/24024378).
Please provide account names so we can enable them for those who want to try.

**Comment by StefanScherer** [comment]
That are great news and the example looks very promising. My AppVeyor account is `StefanScherer` as I login from GitHub.

**Comment by IlyaFinkelshteyn** [comment]
@StefanScherer it should be enabled for you now. Please set image to `Windows Server 2019`.  [Hoe to set image](https://www.appveyor.com/docs/build-environment/#choosing-image-for-your-builds), just in case :)

**Comment by StefanScherer** [comment]
Thanks @IlyaFinkelshteyn it works perfectly! Did my first manifest list build Traefik for 1803 and 1809 images without my hacky rebase-docker-image tool 🎉 
https://github.com/StefanScherer/dockerfiles-windows/pull/403/files

**Comment by IlyaFinkelshteyn** [comment]
@StefanScherer a lot! Knowing it works for you, we feel more confident to advertise it for a wider public :) Hopefully longer than usual `Starting` stage (time to fire an Azure VM) is OK.

**Comment by alangford** [comment]
First of all Thank you so much for adding me this is great! This resolved our .net 2.2 docker image issue.

I am noticing a new issue with our build on the Windows Server 2019 it looks like the mongodb service cannot be found. will this be added in the future or should i look into installing it manually?

**Comment by IlyaFinkelshteyn** [comment]
Welcome :) Yes we plan to add software to this image as needed. For now you can use [this script](https://github.com/appveyor/build-images/blob/master/scripts/Windows/install_mongo.ps1) to install it on the fly.

**Comment by alangford** [comment]
> Welcome :) Yes we plan to add software to this image as needed. For now you can use [this script](https://github.com/appveyor/build-images/blob/master/scripts/Windows/install_mongo.ps1) to install it on the fly.

Thanks again!

**Comment by iameli** [comment]
Hey, this all sounds really great! Any way I could get AppVeyor account `iameli` on the Visual Studio 2019 list? Would really help us out bringing some Docker builds in sync.

**Comment by IlyaFinkelshteyn** [comment]
@iameli sure, enabled. But it is not `Visual Studio 2019`, it is `Windows Server 2019`. We are preparing `Visual Studio 2019` as well, but we are no sure it will be on VMs with nested virtualization and Docker/WSL and all respective images installed. `Windows Server 2019` is the image for all heterogeneous builds.

**Comment by iameli** [comment]
Got it - just looking to run Docker commands so that works perfectly fine. Thanks a lot!

**Comment by IlyaFinkelshteyn** [comment]
@alangford we deployed updated `Windows Server 2019` image with MongoDB.

**Comment by idubnori** [comment]
@IlyaFinkelshteyn Could you add my account `idubnori` to be able to use `Windows Server 2019` image? I'd like to use it to prepare the integration test env as Linux containers on my contributing oss...

**Comment by IlyaFinkelshteyn** [comment]
@idubnori added

**Comment by idubnori** [comment]
@IanBFarrell Thank you so much! worked for me.

**Comment by ericxtang** [comment]
Hi can we add the Windows Server 2019 image to the `ericxtang` appveyor account?

**Comment by IlyaFinkelshteyn** [comment]
@ericxtang done

**Comment by olljanat** [comment]
I'm just curious that when 2019 image should be public?

**Comment by IlyaFinkelshteyn** [comment]
@olljanat we are considering to merge it with `Visual Studio 2019` image which is already public.

**Comment by khs1994** [comment]
@IlyaFinkelshteyn  Can add the `Windows Server 2019` image to the `khs1994-docker` appveyor account,thanks

**Comment by IlyaFinkelshteyn** [comment]
@khs1994 Sure, done.
